### PR TITLE
TKO VMC_ON_AWS_RA

### PR DESCRIPTION
**Pull request by bbharathan**
TKO VMC_ON_AWS_RA

### CODE REVIEW

File: src/reference-designs/tko-on-vmc-aws.md [hunk 1]
```diff
@@ -158,6 +163,19 @@ The virtual services can be spanned across multiple service engines if the assoc
...
 
+### Tanzu Kubernetes Grid Clusters Recommendations
+
+|**Decision ID**|**Design Decision**|**Design Justification**|**Design Implications**|
+| --- | --- | --- | --- |
+|TKO-TKG-001|Deploy TKG management cluster from TKG Installer UI.|Simplified method of installation|NA|
+|TKO-TKG-002|Register the management cluster with Tanzu Mission Control.|Tanzu Mission Control automates the creation of the Tanzu Kubernetes clusters and manages the life cycle of all clusters centrally.|Only Antrea CNI and Photon OS are supported on Workload clusters created from the TMC portal.  |
+|TKO-TKG-003|Use NSX Advanced Load Balancer as your control plane endpoint provider and for application load balancing.|AVI is tightly coupled with TKG and vSphere. Since AVI is a VMware product, customers will have single point of contact for support.| Adds NSX Advanced Load Balancer  License Cost to the solution  |
+|TKO-TKG-004|Deploy Tanzu Kubernetes Management clusters in large form factor|Large form factor should suffice to integrate TKG Mgmt Cluster with TMC, pinniped and velero deployment. This must be capable of accommodating 100+ Tanzu Workload Clusters|Consume more Resources from Infrastructure |
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:173]:
```suggestion
|TKO-TKG-004|Deploy Tanzu Kubernetes Management clusters in large form factor|Large form factor should suffice to integrate TKG Management Cluster with TMC, pinniped and velero deployment. This must be capable of accommodating 100+ Tanzu Workload Clusters|Consume more Resources from Infrastructure |
```

File: src/reference-designs/tko-on-vmc-aws.md [hunk 2]
```diff
@@ -179,31 +197,40 @@ This topology provides the following benefits:
...
 | TKG management VIP/data network | Optional               | Virtual services for all user-managed packages (such as Contour and Harbor) hosted on the shared services cluster.                                                                           |
 | TKG workload VIP/data network   | Optional             | Virtual services for all applications hosted on the workload clusters.                                                                                                                      |
+### Network Recommendations
+
+The key network recommendations for a production-grade Tanzu Kubernetes Grid deployment with NSX-T Data Center Networking are as follows:
+
+|**Decision ID**|**Design Decision**|**Design Justification**|**Design Implications**|
+| --- | --- | --- | --- |
+| TKO-NET-001     | Use separate networks for TKG management and workload clusters.   | To have a flexible firewall and security policies                                         | Sharing the same network for multiple clusters can complicate the creation of firewall rules.                          |
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:215]:
```suggestion
| TKO-NET-001     | Use separate networks for TKG management and workload clusters.   | To have a flexible firewall and security policies | Sharing the same network for multiple clusters can complicate the creation of firewall rules. |
```

File: src/reference-designs/tko-on-vmc-aws.md [hunk 3]
```diff
@@ -179,31 +197,40 @@ This topology provides the following benefits:
...
+| TKO-NET-002     | Use separate networks for workload clusters based on their usage. | Isolate production Kubernetes clusters from dev/test clusters.                            | <p>A separate set of service engines can be used to separate dev/test workload clusters from prod clusters.</p><p></p> |
+| TKO-NET-003     | Configure DHCP for TKG clusters.                                  | Tanzu Kubernetes Grid does not support static IP assignments for Kubernetes VM components | Enable DHCP on the logical segments that are used to host TKG clusters.                                            |
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:217]:
```suggestion
| TKO-NET-003  | Configure DHCP for TKG clusters. | Tanzu Kubernetes Grid does not support static IP assignments for Kubernetes VM components | Enable DHCP on the logical segments that are used to host TKG clusters. |
```

File: src/reference-designs/tko-on-vmc-aws.md [hunk 4]
```diff
@@ -179,31 +197,40 @@ This topology provides the following benefits:
...
-| TKG management network      | TKG-Management     | 192.168.12.1/24  | 192.168.12.2 - 192.168.12.251 | NA                            |
-| TKG workload network        | TKG-Workload  | 192.168.13.1/24  | 192.168.13.2 - 192.168.13.251 | NA                            |
-| TKG cluster VIP network     | TKG-Cluster-VIP    | 192.168.14.1/26  | 192.168.14.2 - 192.168.14.30                            | 192.168.14.31 - 192.168.14.60  |
-| TKG management VIP network        | TKG-Management-VIP         | 192.168.15.1/26  | 192.168.15.2 - 192.168.15.30                            | 192.168.15.31 - 192.168.15.60  |
-| TKG workload VIP network    | TKG-Workload-VIP   | 192.168.16.1/26  | 192.168.16.2 - 192.168.16.30                            | 192.168.16.31 - 192.168.16.60  |
-| TKG shared services network | TKG-Shared-Service | 192.168.17.1/24  | 192.168.17.2 - 192.168.17.251 | NA                              |
+|**Network Type**|**Port Group Name**|**Gateway CIDR**|**DHCP Pool**|**NSX ALB IP Pool**|
+| --- | --- | --- | --- | --- |
+| NSX ALB management network        | `sfo01-w01-vds01-albmanagement`       | 192.168.11.1/27  | 192.168.11.15 - 192.168.11.30 | NA|
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:225]:
```suggestion
| NSX ALB management network | `sfo01-w01-vds01-albmanagement` | 192.168.11.1/27  | 192.168.11.15 - 192.168.11.30 | NA|
```

File: src/reference-designs/tko-on-vmc-aws.md [hunk 5]
```diff
@@ -226,8 +253,8 @@ VMware Cloud on AWS uses a management gateway and compute gateway. These gateway
 
 The following table provides a list of firewall rules based on the assumption that there is no firewall within a subnet/VLAN.
 
-| **Source**                                                                                                                 | **Destination**                                  | **Protocol:Port**               | **Description**                                                                           |
-| ---------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------- | --------------------------------- | ------------------------------------------------------------------------------------------- |
+|**Source**|**Destination**|**Protocol:Port**|**Description**|
+| --- | --- | --- | --- |
 | Client machine                                                                                                             | NSX ALB controller nodes and cluster IP address. | TCP:443                         | To access NSX ALB portal for configuration.                                               |
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:258]:
```suggestion
| Client machine  | NSX ALB controller nodes and cluster IP address. | TCP:443 | To access NSX ALB portal for configuration. |
```

File: src/reference-designs/tko-on-vmc-aws.md [hunk 6]
```diff
@@ -226,8 +253,8 @@ VMware Cloud on AWS uses a management gateway and compute gateway. These gateway
...
 | Client machine                                                                                                             | vCenter Server                                   | TCP:443                         | To create resource pools, VM folders, etc, in vCenter.                                    |
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:259]:
```suggestion
| Client machine | vCenter Server | TCP:443  | To create resource pools, VM folders, etc, in vCenter.  |
```

File: src/reference-designs/tko-on-vmc-aws.md [hunk 7]
```diff
@@ -226,8 +253,8 @@ VMware Cloud on AWS uses a management gateway and compute gateway. These gateway
...
 | <p>Bootstrap machine</p><p></p>                                                                                            | projects.registry.vmware.com                     | TCP:443                         | To pull binaries from VMware public repo for TKG installation.                            |
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:260]:
```suggestion
| Bootstrap machine| projects.registry.vmware.com  | TCP:443   | To pull binaries from VMware public repo for TKG installation. |
```

File: src/reference-designs/tko-on-vmc-aws.md [hunk 8]
```diff
@@ -243,94 +270,65 @@ The following table provides a list of firewall rules based on the assumption th
...
-| <p>TKG management network CIDR</p><p></p><p>TKG shared services network CIDR.</p><p></p><p>TKG workload network CIDR.</p> | <p>Harbor registry</p><p></p><p>(optional)</p>                                                          | TCP:443                     | Allows components to retrieve container images from a local image registry.                                                                                                                                                             |
+|**Source**|**Destination**|**Protocol:Port**|**Description**|
+| --- | --- | --- | --- |
+|TKG management network CIDR</p><p></p><p>TKG shared services network CIDR.</p><p></p><p>TKG workload network CIDR.</p> | <p>Harbor registry</p><p></p><p>(optional)</p>                                                          | TCP:443                     | Allows components to retrieve container images from a local image registry.                                                                                                                                                             |
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:275]:
```suggestion
|TKG management network CIDR</p><p></p><p>TKG shared services network CIDR.</p><p></p><p>TKG workload network CIDR.</p> | <p>Harbor registry</p><p></p><p>(optional)</p>| TCP:443  | Allows components to retrieve container images from a local image registry.  |
```

File: src/reference-designs/tko-on-vmc-aws.md [hunk 9]
```diff
@@ -243,94 +270,65 @@ The following table provides a list of firewall rules based on the assumption th
...
 | <p>Client machine</p><p></p>                                                                                              | <p>console.cloud.vmware.com</p><p>\*.tmc.cloud.vmware.com</p><p></p><p>projects.registry.vmware.com</p> | TCP:443                     | <p>To access cloud Services portal to configure networks in VMC SDDC.</p><p>To access the TMC portal for TKG clusters registration and other SaaS integration.</p><p>To pull binaries from VMware public repo for TKG installation.</p> |
 | Client machine                                                                                                            | <p>TKG management VIP range.</p><p>TKG Workload VIP range.</p>                                          | <p>TCP:80</p><p>TCP:443</p> | To http/https workloads in shared services and workload cluster.                                                                                                                                                                        |
```
Comment (reshmarajkumar-vmw) [inline src/reference-designs/tko-on-vmc-aws.md:277]:
```suggestion
| Client machine  | <p>TKG management VIP range.</p><p>TKG Workload VIP range.</p> | <p>TCP:80</p><p>TCP:443</p> | To http/https workloads in shared services and workload cluster. |
```

**Review by reshmarajkumar-vmw — COMMENTED** [review]
Please find comments inline

**Review by reshmarajkumar-vmw — APPROVED** [review]
Content reviewed and finalized over call

**Comment by smuthukumar-vmw** [comment]
Checks are queued and not run.

**Comment by smuthukumar-vmw** [comment]
Reopening for checks to complete

**Pull request merged**